### ISSUE TITLE
Photon appears to ignore 'jetpack_photon_skip_image'

### ISSUE THREAD

**Issue by rothschild86** [Automattic/jetpack#4626]
Followed advice at https://developer.jetpack.com/hooks/jetpack_photon_skip_image/

to exclude images with a certain string in a name. Noticed that it does not appear to have effect.

```
function my_photon_exception( $val, $src, $tag ) {
        if ( $src == 'YOUR_IMAGE_URL' ) {
                return true;
        }
        return $val;
}
add_filter( 'jetpack_photon_skip_image', 'my_photon_exception', 10, 3 );
```

decided to short circuit the code to just:

```
function my_photon_exception( $val, $src, $tag ) {
return true;
}
add_filter( 'jetpack_photon_skip_image', 'my_photon_exception', 10, 3 );
```

Was surprised to find out that all images were still being pulled from photon.

Is this hook obsolete by any chance?

Thank you for looking into this!

**Comment by jeherve** [comment]
The filter still works, but I found that it wasn't apply to `srcset` URLs. I've just submitted #4629 to take care of this.

Thanks for the report!

**Comment by rothschild86** [comment]
Ah, good!

**Comment by rothschild86** [comment]
ah man... another milestone slip... really looking forward to reenabling photon :) I know you guys are cranking it out. Just my 2 cents :)

**Comment by jeherve** [comment]
@rothschild86 Feel free to give the patch a try on your site, and let us know how it goes!

**Comment by kraftbj** [comment]
@rothschild86 Just for context on the milestone slip, 4.2.1 was the original maintenance release planned for 4.2. We had an issue with 4.2 impacting really really large multisites (2M+ users), so we pulled back 4.2, pushed back everything originally meant for 4.2.1 and used 4.2.1 to re-release 4.2. So, then 4.2.2 became the maintenance release which was going fine until Facebook changed some of their API responses without notification, so we rushed 4.2.2 out with what was ready to go since it impacted something that generates a high support load typically.

All that said, the milestone is slipping, but more because fires were sparked than this issue being ignored or overlooked. :)

Please do test out the patch though and help us confirm it is resolving what you're reporting.

**Comment by rothschild86** [comment]
Ah, that sounds like no fun.

Added the two changes, and that broke my snippet:

```
//all images that contain -nocdn in their name will not go thru photon
function my_photon_exception( $val, $src, $tag ) {  
        if ( strpos($src, 'nocdn') !== false ) {
                return true;
        }       
  return $val;
}
add_filter( 'jetpack_photon_skip_image', 'my_photon_exception', 10, 3 );
```

Warning: Missing argument 3 for my_photon_exception() in /home/unikur5/public_html/rothschildportraiture/wp-content/plugins/code-snippets/php/snippet-ops.php(383) : eval()'d code on line 3

As to why, no idea... Seems legit. Not too strong on php here.

**Comment by jeherve** [comment]
@rothschild86 nice catch. This should be fixed by cf62e03765919e0d65ee0dc7bea3332960f796f1

**Comment by rothschild86** [comment]
yep, now we are in business. confirmed working.
